File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/event/LayoutPassReason.java
package io.github.orioncraftmc.meditate.internal.event;

public enum LayoutPassReason //Type originates from: event.h
{
    kInitial(0),
    kAbsLayout(1),
    kStretch(2),
    kMultilineStretch(3),
    kFlexLayout(4),
    kMeasureChild(5),
    kAbsMeasureChild(6),
    kFlexMeasure(7),
    COUNT(8);

    public static final int SIZE = java.lang.Integer.SIZE;
    private static java.util.HashMap<Integer, LayoutPassReason> mappings;
    private final int intValue;

    LayoutPassReason(int value) {
        intValue = value;
        getMappings().put(value, this);
    }

    private static java.util.HashMap<Integer, LayoutPassReason> getMappings() {
        if (mappings == null) {
            synchronized (LayoutPassReason.class) {
                if (mappings == null) {
                    mappings = new java.util.HashMap<>();
                }
            }
        }
        return mappings;
    }

    public static LayoutPassReason forValue(int value) {
        return getMappings().get(value);
    }

    public int getValue() {
        return intValue;
    }
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/event/LayoutPassReason.kt
package app.cash.redwood.yoga.event

enum class LayoutPassReason //Type originates from: event.h
  (private val intValue: Int) {
  kInitial(0),
  kAbsLayout(1),
  kStretch(2),
  kMultilineStretch(3),
  kFlexLayout(4),
  kMeasureChild(5),
  kAbsMeasureChild(6),
  kFlexMeasure(7),
  COUNT(8);

  fun getValue(): Int {
    return intValue
  }

  companion object {
    private val mappings: Map<Int, LayoutPassReason> = values().associateBy { it.intValue }

    private fun getMappings(): Map<Int, LayoutPassReason> {
      return mappings
    }

    fun forValue(value: Int): LayoutPassReason? {
      return getMappings()[value]
    }
  }
}
